src/roots.jl

```diff
@@ -1,24 +1,84 @@
 
-import IntervalArithmetic: diam, isinterior, bisect, isnan
+import IntervalArithmetic: diam, bisect, isnai
 
 export branch_and_prune, Bisection, Newton
 
-diam(r::Root) = diam(interval(r))
-isnan(X::IntervalBox) = any(isnan.(X))
-isnan(r::Root) = isnan(interval(r))
-
-struct RootProblem{T}
+struct RootProblem{C, F, G, R, S, T}
+    contractor::Type{C}
+    f::F
+    derivative::G
+    region::R
+    search_order::Type{S}
     abstol::T
+    reltol::T  # TODO
+    max_iteration::Int  # TODO
```

Comment:
What is the meaning of "TODO"?

Reply:
They options are straight up not used currently 😬.

I've moved this to the docstrings.